m5_forecasting/data_processor.py

```
@@ -0,0 +1,156 @@
+import pandas as pd
+import numpy as np
+import logging
+
+# Set up logging
+logger = logging.getLogger(__name__)
+
+class DataProcessor:
+    def __init__(self, config):
+        """Initialize the DataProcessor with configuration details."""
+        required_keys = ['sales_filepath', 'calendar_filepath', 'sell_prices_filepath', 'horizon']  
```

Ruff

Trailing whitespace

Remove trailing whitespace

(W291)